### ISSUE TITLE
problems with shiny deployment

### ISSUE THREAD

**Issue by edgarjavi** [hojsgaard/gRain#5]
Dear Søren,

I don't know if this the appropriate space to ask this question, but I am trying to deploy a shiny app that uses gRain as part of the workflow, however, I haven't been able to publish the app because I get the error

Error: Unhandled Exception: Child Task 769581942 failed: Error parsing manifest: Unable to determine package source for Bioconductor package gRbase: Repository must be specified
Execution halted

I saw in comments for another package that the problem was the interdependency between Bioconductor dependencies and the CRAN package.

Do you have any hints that can help fix this problem, I appreciate any help you can provide me.

Best,

Edgar

**Comment by hojsgaard** [comment]
gRbase is a cran package - not bioconductor.
Best Søren



Sendt fra min Samsung Galaxy-smartphone.



-------- Oprindelig besked --------
Fra: Javier <notifications@github.com>
Dato: 31.07.2020 17.05 (GMT+01:00)
Til: hojsgaard/gRain <gRain@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Emne: [hojsgaard/gRain] problems with shiny deployment (#5)


Dear Søren,

I don't know if this the appropriate space to ask this question, but I am trying to deploy a shiny app that uses gRain as part of the workflow, however, I haven't been able to publish the app because I get the error

Error: Unhandled Exception: Child Task 769581942 failed: Error parsing manifest: Unable to determine package source for Bioconductor package gRbase: Repository must be specified
Execution halted

I saw in comments for another package that the problem was the interdependency between Bioconductor dependencies and the CRAN package.

Do you have any hints that can help fix this problem, I appreciate any help you can provide me.

Best,

Edgar

**Comment by edgarjavi** [comment]
Yes, and I don't understand why it is looking in bioconductor, the only explanation I can give to that is that there are dependencies from Bioconductor and thus it thinks it needs the bioconductor package. I tried then installing in using the source packages I did
devtools::install_github("hojsgaard/gRbase")
devtools::install_github("hojsgaard/gRain")

and that seem to have worked but it didn't finished the compilaton of gRain and gave me this error

Error: package or namespace load failed for ‘gRain’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/opt/R/4.0.2/lib/R/library/00LOCK-hojsgaard-gRain-0b39e1e/00new/gRain/libs/gRain.so':
  libgfortran.so.4: cannot open shared object file: No such file or directory
Error: loading failed

I have tried all possible advice from the web but I seem to be stuck, and this is the only package that need to install to run my shiny application, any help would be greatly appreciated.

gRain runs perfectly on Rstudio, as well as in the shiny app, the only think that I have problems with is while deploying the shiny app that fails the installation.

Thanks

Edgar

**Comment by hojsgaard** [comment]
I have had biocViews in the DESCRIPTION file. This has now been removed. Check new version from github.

Best Søren

________________________________
Fra: Javier <notifications@github.com>
Sendt: 12. august 2020 03:39:35
Til: hojsgaard/gRain
Cc: Søren Højsgaard; Comment
Emne: Re: [hojsgaard/gRain] problems with shiny deployment (#5)


Yes, and I don't understand why it is looking in bioconductor, the only explanation I can give to that is that there are dependencies from Bioconductor and thus it thinks it needs the bioconductor package. I tried then installing in using the source packages I did
devtools::install_github("hojsgaard/gRbase")
devtools::install_github("hojsgaard/gRain")

and that seem to have worked but it didn't finished the compilaton of gRain and gave me this error

Error: package or namespace load failed for ‘gRain’ in dyn.load(file, DLLpath = DLLpath, ...):
unable to load shared object '/opt/R/4.0.2/lib/R/library/00LOCK-hojsgaard-gRain-0b39e1e/00new/gRain/libs/gRain.so':
libgfortran.so.4: cannot open shared object file: No such file or directory
Error: loading failed

I have tried all possible advice from the web but I seem to be stuck, and this is the only package that need to install to run my shiny application, any help would be greatly appreciated.

Thanks

Edgar
